### ISSUE TITLE
Version 5.1.35 of jdbc-mysql breaks warbler jars

### ISSUE THREAD

**Issue by camertron** [jruby/activerecord-jdbc-adapter#654]
I'm running sidekiq outside of rails, JRuby version 1.7.15. If I lock jdbc-mysql to v5.1.33 and build a jar using warbler, everything works perfectly. If I instead use v5.1.35, all hell breaks loose with this stack trace:

```
Could not initialize class com.mysql.jdbc.Util (Java::JavaLang::NoClassDefFoundError)
com.mysql.jdbc.NonRegisteringDriver.parseURL(com/mysql/jdbc/NonRegisteringDriver.java:730)
com.mysql.jdbc.NonRegisteringDriver.connect(com/mysql/jdbc/NonRegisteringDriver.java:316)
java.lang.reflect.Method.invoke(java/lang/reflect/Method.java:483)
RUBY.connection(/Users/cameron/workspace/lumos-rosette/queue-worker/queue-worker.jar!/gems/activerecord-jdbc-adapter-1.3.16/lib/arjdbc/jdbc/driver.rb:39)
RUBY.newConnection(/Users/cameron/workspace/lumos-rosette/queue-worker/queue-worker.jar!/gems/activerecord-jdbc-adapter-1.3.16/lib/arjdbc/jdbc/connection.rb:114)
ActiveRecord$$ConnectionAdapters$$JdbcConnection$$JdbcConnectionFactoryImpl_80659890.newConnection(ActiveRecord$$ConnectionAdapters$$JdbcConnection$$JdbcConnectionFactoryImpl_80659890.gen:13)
ActiveRecord$$ConnectionAdapters$$JdbcConnection$$JdbcConnectionFactoryImpl_80659890.newConnection(ActiveRecord$$ConnectionAdapters$$JdbcConnection$$JdbcConnectionFactoryImpl_80659890.gen:13)
arjdbc.jdbc.RubyJdbcConnection.newConnection(arjdbc/jdbc/RubyJdbcConnection.java:1509)
arjdbc.mysql.MySQLRubyJdbcConnection.newConnection(arjdbc/mysql/MySQLRubyJdbcConnection.java:237)
arjdbc.jdbc.RubyJdbcConnection.initConnection(arjdbc/jdbc/RubyJdbcConnection.java:458)
arjdbc.jdbc.RubyJdbcConnection.init_connection(arjdbc/jdbc/RubyJdbcConnection.java:450)
RUBY.initialize(/Users/cameron/workspace/lumos-rosette/queue-worker/queue-worker.jar!/gems/activerecord-jdbc-adapter-1.3.16/lib/arjdbc/jdbc/connection.rb:15)
RUBY.initialize(/Users/cameron/workspace/lumos-rosette/queue-worker/queue-worker.jar!/gems/activerecord-jdbc-adapter-1.3.16/lib/arjdbc/jdbc/adapter.rb:68)
RUBY.jdbc_connection(/Users/cameron/workspace/lumos-rosette/queue-worker/queue-worker.jar!/gems/activerecord-jdbc-adapter-1.3.16/lib/arjdbc/jdbc/connection_methods.rb:12)
RUBY.mysql_connection(/Users/cameron/workspace/lumos-rosette/queue-worker/queue-worker.jar!/gems/activerecord-jdbc-adapter-1.3.16/lib/arjdbc/mysql/connection_methods.rb:74)
RUBY.new_connection(/Users/cameron/workspace/lumos-rosette/queue-worker/queue-worker.jar!/gems/activerecord-3.2.21/lib/active_record/connection_adapters/abstract/connection_pool.rb:315)
RUBY.checkout_new_connection(/Users/cameron/workspace/lumos-rosette/queue-worker/queue-worker.jar!/gems/activerecord-3.2.21/lib/active_record/connection_adapters/abstract/connection_pool.rb:325)
RUBY.checkout(/Users/cameron/workspace/lumos-rosette/queue-worker/queue-worker.jar!/gems/activerecord-3.2.21/lib/active_record/connection_adapters/abstract/connection_pool.rb:247)
org.jruby.RubyKernel.loop(org/jruby/RubyKernel.java:1511)
RUBY.checkout(/Users/cameron/workspace/lumos-rosette/queue-worker/queue-worker.jar!/gems/activerecord-3.2.21/lib/active_record/connection_adapters/abstract/connection_pool.rb:242)
RUBY.mon_synchronize(/var/folders/ls/3sjzmycd3h5dpsp_fs9_8zjr0000gp/T/jruby7013762319390601087extract/jruby-stdlib-1.7.20.jar!/META-INF/jruby.home/lib/ruby/1.9/monitor.rb:211)
RUBY.checkout(/Users/cameron/workspace/lumos-rosette/queue-worker/queue-worker.jar!/gems/activerecord-3.2.21/lib/active_record/connection_adapters/abstract/connection_pool.rb:239)
RUBY.connection(/Users/cameron/workspace/lumos-rosette/queue-worker/queue-worker.jar!/gems/activerecord-3.2.21/lib/active_record/connection_adapters/abstract/connection_pool.rb:102)
RUBY.mon_synchronize(/var/folders/ls/3sjzmycd3h5dpsp_fs9_8zjr0000gp/T/jruby7013762319390601087extract/jruby-stdlib-1.7.20.jar!/META-INF/jruby.home/lib/ruby/1.9/monitor.rb:211)
RUBY.connection(/Users/cameron/workspace/lumos-rosette/queue-worker/queue-worker.jar!/gems/activerecord-3.2.21/lib/active_record/connection_adapters/abstract/connection_pool.rb:101)
RUBY.with_connection(/Users/cameron/workspace/lumos-rosette/queue-worker/queue-worker.jar!/gems/activerecord-3.2.21/lib/active_record/connection_adapters/abstract/connection_pool.rb:129)
RUBY.with_connection(/Users/cameron/workspace/lumos-rosette/queue-worker/queue-worker.jar!/bundler/gems/rosette-active-record-6a5c331dc453/lib/rosette/data_stores/active_record_data_store.rb:326)
RUBY.lookup_commit_log(/Users/cameron/workspace/lumos-rosette/queue-worker/queue-worker.jar!/bundler/gems/rosette-active-record-6a5c331dc453/lib/rosette/data_stores/active_record_data_store.rb:113)
RUBY.lookup_commit_log(/Users/cameron/workspace/lumos-rosette/queue-worker/queue-worker.jar!/bundler/gems/rosette-core-08313a1e1579/lib/rosette/queuing/commits/commit_job.rb:85)
RUBY.find_or_create_commit_log(/Users/cameron/workspace/lumos-rosette/queue-worker/queue-worker.jar!/bundler/gems/rosette-core-08313a1e1579/lib/rosette/queuing/commits/commit_job.rb:73)
RUBY.work(/Users/cameron/workspace/lumos-rosette/queue-worker/queue-worker.jar!/bundler/gems/rosette-core-08313a1e1579/lib/rosette/queuing/commits/commit_job.rb:64)
RUBY.perform(/Users/cameron/workspace/lumos-rosette/queue-worker/queue-worker.jar!/bundler/gems/rosette-queue-sidekiq-602606e9c0b9/lib/rosette/queuing/sidekiq-queue/job_wrapper.rb:17)
RUBY.execute_job(/Users/cameron/workspace/lumos-rosette/queue-worker/queue-worker.jar!/gems/sidekiq-3.3.4/lib/sidekiq/processor.rb:75)
RUBY.process(/Users/cameron/workspace/lumos-rosette/queue-worker/queue-worker.jar!/gems/sidekiq-3.3.4/lib/sidekiq/processor.rb:52)
RUBY.invoke(/Users/cameron/workspace/lumos-rosette/queue-worker/queue-worker.jar!/gems/sidekiq-3.3.4/lib/sidekiq/middleware/chain.rb:127)
RUBY.call(/Users/cameron/workspace/lumos-rosette/queue-worker/queue-worker.jar!/gems/sidekiq-3.3.4/lib/sidekiq/middleware/server/active_record.rb:6)
RUBY.invoke(/Users/cameron/workspace/lumos-rosette/queue-worker/queue-worker.jar!/gems/sidekiq-3.3.4/lib/sidekiq/middleware/chain.rb:129)
RUBY.call(/Users/cameron/workspace/lumos-rosette/queue-worker/queue-worker.jar!/gems/sidekiq-3.3.4/lib/sidekiq/middleware/server/retry_jobs.rb:74)
RUBY.invoke(/Users/cameron/workspace/lumos-rosette/queue-worker/queue-worker.jar!/gems/sidekiq-3.3.4/lib/sidekiq/middleware/chain.rb:129)
RUBY.call(/Users/cameron/workspace/lumos-rosette/queue-worker/queue-worker.jar!/gems/sidekiq-3.3.4/lib/sidekiq/middleware/server/logging.rb:15)
RUBY.with_context(/Users/cameron/workspace/lumos-rosette/queue-worker/queue-worker.jar!/gems/sidekiq-3.3.4/lib/sidekiq/logging.rb:24)
RUBY.call(/Users/cameron/workspace/lumos-rosette/queue-worker/queue-worker.jar!/gems/sidekiq-3.3.4/lib/sidekiq/middleware/server/logging.rb:11)
RUBY.invoke(/Users/cameron/workspace/lumos-rosette/queue-worker/queue-worker.jar!/gems/sidekiq-3.3.4/lib/sidekiq/middleware/chain.rb:129)
org.jruby.RubyProc.call(org/jruby/RubyProc.java:271)
RUBY.invoke(/Users/cameron/workspace/lumos-rosette/queue-worker/queue-worker.jar!/gems/sidekiq-3.3.4/lib/sidekiq/middleware/chain.rb:132)
RUBY.process(/Users/cameron/workspace/lumos-rosette/queue-worker/queue-worker.jar!/gems/sidekiq-3.3.4/lib/sidekiq/processor.rb:51)
RUBY.stats(/Users/cameron/workspace/lumos-rosette/queue-worker/queue-worker.jar!/gems/sidekiq-3.3.4/lib/sidekiq/processor.rb:98)
RUBY.process(/Users/cameron/workspace/lumos-rosette/queue-worker/queue-worker.jar!/gems/sidekiq-3.3.4/lib/sidekiq/processor.rb:50)
org.jruby.RubyKernel.public_send(org/jruby/RubyKernel.java:1962)
RUBY.dispatch(/Users/cameron/workspace/lumos-rosette/queue-worker/queue-worker.jar!/gems/celluloid-0.16.0/lib/celluloid/calls.rb:26)
RUBY.dispatch(/Users/cameron/workspace/lumos-rosette/queue-worker/queue-worker.jar!/gems/celluloid-0.16.0/lib/celluloid/calls.rb:122)
RUBY.invoke(/Users/cameron/workspace/lumos-rosette/queue-worker/queue-worker.jar!/gems/celluloid-0.16.0/lib/celluloid/cell.rb:60)
RUBY.task(/Users/cameron/workspace/lumos-rosette/queue-worker/queue-worker.jar!/gems/celluloid-0.16.0/lib/celluloid/cell.rb:71)
RUBY.task(/Users/cameron/workspace/lumos-rosette/queue-worker/queue-worker.jar!/gems/celluloid-0.16.0/lib/celluloid/actor.rb:357)
RUBY.initialize(/Users/cameron/workspace/lumos-rosette/queue-worker/queue-worker.jar!/gems/celluloid-0.16.0/lib/celluloid/tasks.rb:57)
RUBY.create(/Users/cameron/workspace/lumos-rosette/queue-worker/queue-worker.jar!/gems/celluloid-0.16.0/lib/celluloid/tasks/task_fiber.rb:15)
java.util.concurrent.ThreadPoolExecutor.runWorker(java/util/concurrent/ThreadPoolExecutor.java:1142)
java.util.concurrent.ThreadPoolExecutor$Worker.run(java/util/concurrent/ThreadPoolExecutor.java:617)
java.lang.Thread.run(java/lang/Thread.java:745)
```

I've been able to determine that the mysql connector jar isn't getting loaded because if I add the mysql-connector-java jar to my Jarfile (via maven/jbundler), everything works again.

So the real question is this: what changed between v5.1.33 and v5.1.35? I've compared the two gems and I can't see any difference that would cause an error like this. Any ideas?

**Comment by kares** [comment]
than it mus be an issue with JRuby ... could you please try latest (1.7.20) if the issue still persists?

**Comment by camertron** [comment]
Thanks for the fast response @kares. I tried with JRuby 1.7.20 and jdbc-mysql 5.1.33 and the problem still persists. For now I can just use 5.1.33, so it's fine. Also, what happened to v5.1.34?

**Comment by kares** [comment]
@camertron in that case please try to come up with an app where this can be reproduced, thanks!
jdbc-mysql v5.1.34 simply was not released ... we do not follow each and very release closely.

**Comment by kares** [comment]
just released jdbc-mysql **5.1.36** ... please try if it's still an issue. have no ideas about what it could be, but knowing whether a new versions fails or not is still valuable ... thanks!

**Comment by kares** [comment]
closing as there's been no response - assuming fixed in jdbc-mysql 5.1.36
